### ISSUE TITLE
Validate the format of `expose`

### ISSUE THREAD

**Issue by pedro-nonfree** [docker/compose#2406]
The other day I expend some hours trying to investigate why my docker container was not exposing some ports. And understanding that was a docker problem, and not an app problem.

In dockerfile there is **expose** to say the ports, but **ports** in docker-compose. Ok, there is a change.

But why cannot say that "warning, expose syntax is not valid in docker-compose, you meant ports?" or something like this?

**Comment by dnephin** [comment]
Do you have an example of the config that was invalid but didn't raise an error?

Normally you should get a validation error when the config is incorrect.

**Comment by pedro-nonfree** [comment]
```
pdp-service:
    #image: bitergia/authzforce:4.2.0
    build: bitergia-pdp
    container_name: pdp-container
    hostname: pdp-container
    expose:
        - "8080:8080"

idm-service:
    #image: bitergia/idm-keyrock:4.3.0
    build: bitergia-idm
    container_name: idm
    hostname: idm
    hostname: idm
    links:
        - pdp-service
    volumes:
        - /config
    expose:
        - "5000:5000"
        - "443:443"
    environment:
         - AUTHZFORCE_HOSTNAME=pdp-service
         - AUTHZFORCE_PORT=8080 
         - IDM_KEYROCK_HOSTNAME=idm
         - IDM_KEYROCK_PORT=443
         - MAGIC_KEY=daf26216c5434a0a80f392ed9165b3b4
         - WORKON_HOME=/opt/virtualenvs
         - APP_NAME="FIWAREdevGuide"
         - KEYSTONE_DB=/opt/fi-ware-idm/keystone/keystone.db
         - CONFIG_FILE=/config/idm2chanchan.json
         - PROVISION_FILE=/config/keystone_provision.py
         - KEYSTONE_VERBOSE=no

pep-service:
    #image: bitergia/pep-wilma:4.3.0
    build: bitergia-pep
    container_name: pep-container
    hostname: pep-container
    links:
#        - app
        - idm-service
        - pdp-service
    volumes_from:
        - idm-service:rw
    expose:
        - "1026:1026" #original is 8080
    environment:
        - AUTHZFORCE_HOSTNAME=pdp-service
        - AUTHZFORCE_PORT=8080 
        - IDM_KEYSTONE_HOSTNAME=idm-service
        - IDM_KEYSTONE_PORT=5000
#        - APP_HOSTNAME=app
        - APP_PORT=1026
        - PEP_USERNAME=pepproxy@test.com
        - PEP_PASSWORD=test
        - PEP_PORT=1026
        - IDM_USERNAME=user0@test.com
        - IDM_USERPASS=test
        - MAGIC_KEY=daf26216c5434a0a80f392ed9165b3b4
```

docker ps of idm container (port part) says:
```443/tcp, 5000/tcp```

instead of:
```0.0.0.0:443->443/tcp, 0.0.0.0:5000->5000/tcp```

changing from expose to ports it works as expected. no validation error.

docker-compose version: 1.5.0

**Comment by aanand** [comment]
Fix is in #2464.
